algorithms/Java/searching/ternery-search.java

```diff
@@ -0,0 +1,142 @@
+// Java program to illustrate
+// the iterative approach to ternary search
+// 
+  
+class main {
+  
+    // Function to perform Ternary Search
+    // recursive Function
+    // Time Complexity: O(log3n)
+    // Auxiliary Space: O(log3n)
+  
+    static int ternarySearch(int l, int r, int key, int ar[])
```

Comment:
change the name of the function , there is a function already present with the same name
```suggestion
    static int ternarySearchRecursive(int l, int r, int key, int ar[])
```